### ISSUE TITLE
django channels worker do not see some of expired messages

### ISSUE THREAD

**Issue by ollamh** [proofit404/asgi_rabbitmq#5]
# TL;DR

I'm using custom logic for dealing with expired messages from channel. To do that, I'm using custom class implementation inherited from `asgi_rabbitmq.core.Protocol` class. I do not receive about half of expired messages in `on_dead_letter` method.

# More details

I see that although I have only one worker consumer launched to process requests for websockets (and I believe it's only one that should consume from `dead-letters` queue), in Rabbitmq I can see more than one consumer subscribed to `dead-letters` after I sent some messages using syntax `Group('exchange_name').send(<message>)`. I think it might be the case that some messages are consumed by other connections instead of going to channels worker.

The list of connections is:
* django channels worker (I believe this is fine)
* django daphne (UPD. I missed the fact that they run the same codebase, so, it is fine)
* n processes of django project uwsgi worker (I believe they shouldn't consume from the queue)

# How to reproduce
## Packages versions
```
Python 2.7.6
channels==1.1.3
daphne==1.2.0
asgi-rabbitmq==0.4.1
asgi-redis==1.3.0
asgiref==1.1.1
Django==1.8.13
Twisted==16.6.0
```
## Layer config
```
CHANNEL_LAYERS = {
    "default": {
        'BACKEND': 'my_proj.my_app.asgi_chat_rabbitmq.ChatRabbitmqChannelLayer',
        "CONFIG": {
            "url": os.environ.get('QK_RABBITMQ_URL', 'amqp://guest:guest@localhost:5672/%2F'), <-- in staging it is Cloudamqp url
            'expiry': 10
        },
        "ROUTING": "my_proj.routing.channel_routing",
    },
}
```

## Steps
<s>1. You need to run django uwsgi process as well. </s>I have some logging messages in `on_dead_letter` handler (filtering out `maxlen` reason)
2. Start `python manage.py runserver`
3. Run `python manage.py shell` 
4. Declare Channel/Group in Rabbitmq and send few messages to this group or channel using `Group().send()` syntax from the shell.
5.  In Rabbitmq `dead-letters` queue you will see more than one consumer for this queue. (1 for runserver and 1 from django shell)
6. Some messages will appear in log of runserver process, other will appear in django shell.

Thanks!

**Comment by ollamh** [comment]
@proofit404 There is no loss in messages, it was my implementation error, but additional connections appear anyway, which is a bit disturbing, but since it doesn't affect functionality, I will close it. Anyway, thanks for the quick response!

**Comment by proofit404** [comment]
Connections to the broker are created in the layer constructor. So if the process creates an instance of the layer, you will see RabbitMQ connections. Also, this process will handle dead lettered messages.
